File: docs/vs-2015/snippets/visualbasic/VS_Snippets_VSSDK/vssdksupportprojectconfigurationproperties/vb/myprojectpropertypage.vb
Imports System.ComponentModel

Public Class MyProjectPropertyPage
    '<Snippet2>
    Private _myConfigProp As String
    Private _isDirty As Boolean

    Public Property IsDirty As Boolean
        Get
            Return _isDirty
        End Get
        Set(ByVal value As Boolean)
            _isDirty = value
        End Set
    End Property

    <Category("My Category")> _
    <DisplayName("My Config Property")> _
    <Description("My Description")> _
    Public Property MyConfigProp() As String
        Get
            Return _myConfigProp
        End Get
        Set(ByVal value As String)
            _myConfigProp = value
            IsDirty = True
        End Set
    End Property
    '</Snippet2>
End Class

File: docs/vs-2015/snippets/csharp/VS_Snippets_VSSDK/vssdksupportprojectconfigurationproperties/cs/myprojectpropertypage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace Microsoft.VSSDKSupportProjectConfigurationProperties
{
    public class MyProjectPropertyPage
    {
        //<Snippet2>
        public bool IsDirty { get; set; }
        private string myConfigProp;

        [Category("My Category")]
        [DisplayName("My Config Property")]
        [Description("My Description")]
        public string MyConfigProp
        {
            get { return myConfigProp; }
            set { myConfigProp = value; IsDirty = true; }
        }
        //</Snippet2>
    }
}
